tensorboard/webapp/metrics/views/card_renderer/utils.ts

```diff
@@ -96,45 +96,32 @@ export function maybeClipLinkedTimeSelection(
   minStep: number,
   maxStep: number
 ): TimeSelectionView {
-  if (
-    // Case when timeSelection contains extents.
-    (timeSelection.start.step <= minStep &&
-      timeSelection.end &&
-      maxStep <= timeSelection.end.step) ||
-    // Case when start of timeSelection is within extent.
-    (minStep <= timeSelection.start.step &&
-      timeSelection.start.step <= maxStep) ||
-    // Case when end of timeSelection is within extent.
-    (timeSelection.end &&
-      minStep <= timeSelection.end?.step &&
-      timeSelection.end?.step <= maxStep)
-  ) {
-    return {
-      startStep: timeSelection.start.step,
-      endStep: timeSelection.end?.step ?? null,
-      clipped: false,
-    };
-  }
-
-  // When time selection and data extent (in step axis) do not overlap,
-  // default single select min or max data extent depending on which side
-  // the time selection is at.
-
-  // Case when time selection is on the right of the maximum of the
-  // time series.
-  if (maxStep <= timeSelection.start.step) {
-    return {
-      startStep: maxStep,
-      endStep: null,
-      clipped: true,
-    };
-  }
-  // Case when time selection is on the left of the minimum of the time
-  // series.
+  /**
+   * 1) Take greatest between the minStep and the current step.
+   *      This ensures that the current step is >= minStep
+   * 2) Take the minimum between the output of step 1 and the maxStep.
+   *      This ensures the result is <= maxStep.
+   */
+  const startStep = Math.min(
+    Math.max(minStep, timeSelection.start.step),
+    maxStep
+  );
+  /**
+   * 1) Take smallest between the maxStep and the current step.
+   *      This ensures that the current step is <= maxStep
+   * 2) Take the greatest between the output of step 1 and the minStep.
+   *      This ensures the result is >= minStep.
+   */
+  const endStep = timeSelection.end
+    ? Math.max(Math.min(maxStep, timeSelection.end.step), minStep)
```

Comment:
Does this equivalent to 
`Math.min(Math.max(minStep, timeSelection.end.step), maxStep)?`

Reply:
Yes. I'm sensing that these nested `Math.min` / `Math.max` calls are not as readable as I had hoped. I'm going to abstract this logic to a helper function to improve readability.